src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/multiples/MultipleReferenceService.java

```diff
@@ -0,0 +1,239 @@
+        } catch (RestClientResponseException e) {
+            throw (CaseCreationException)
+                    new CaseCreationException(String.format("%s with %s", errorMessage, e.getMessage())).initCause(e);
+        }
+    }
+
+    public void updateMultipleLegalRepCollection(
+            String userToken,
+            String caseTypeId,
+            String jurisdiction,
+            MultipleData multiDataToUpdate,
+            String multipleId,
+            String caseRef,
+            String legalRepId) {
+
+        ListTypeItem<SubCaseLegalRepDetails> legalRepCollection = multiDataToUpdate.getLegalRepCollection();
+        if (legalRepCollection == null) {
+            legalRepCollection = new ListTypeItem<>();
+            multiDataToUpdate.setLegalRepCollection(legalRepCollection);
+        }
+
+        boolean caseExists = false;
+        for (GenericTypeItem<SubCaseLegalRepDetails> details : legalRepCollection) {
+            if (details.getValue().getCaseReference().equals(caseRef)) {
+                caseExists = true;
+                if (details.getValue().getLegalRepIds().stream().noneMatch(
+                        item -> legalRepId.equals(item.getValue()))) {
+                    GenericTypeItem<String> legalRepList = GenericTypeItem.from(legalRepId);
+                    details.getValue().getLegalRepIds().add(legalRepList);
+                } else {
+                    log.info("Legal Rep already exists in collection");
+                    return;
+                }
+                break;
+            }
+        }
+
+        if (!caseExists) {
+            ListTypeItem<String> newLegalRepList = ListTypeItem.from(legalRepId);
+            GenericTypeItem<SubCaseLegalRepDetails> newDetails =
+                    GenericTypeItem.from(new SubCaseLegalRepDetails(caseRef, newLegalRepList));
+            legalRepCollection.add(newDetails);
+        }
+
+        multiDataToUpdate.setLegalRepCollection(legalRepCollection);
+        multipleCasesSendingService.sendUpdateToMultiple(
+                userToken, caseTypeId, jurisdiction, multiDataToUpdate, multipleId);
+    }
+
+    private String buildQueryForGetMultipleByReference(String multipleReference) {
+        TermsQueryBuilder termsQueryBuilder = termsQuery(MULTIPLE_CASE_REFERENCE_KEYWORD, multipleReference);
+
+        return new SearchSourceBuilder()
+                .size(MAX_ES_SIZE)
+                .query(termsQueryBuilder)
+                .toString();
+    }
+
+    private HttpHeaders createHeaders(String serviceAuthorizationToken, String accessToken) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+        headers.set(HttpHeaders.AUTHORIZATION, accessToken);
+        headers.set(SERVICE_AUTHORIZATION, serviceAuthorizationToken);
+        return headers;
+    }
```

Comment:
is it worth making the one in CcdCaseAasignment public so you can reference it rather than duplicating it?